src/core/ledger/policies/immediate.js

```diff
@@ -22,16 +22,33 @@ export type ImmediatePolicy = {|
 
 /**
  * Split a grain budget in proportion to the cred scores in
- * the most recent time interval.
+ * the most recent time interval, with the option to extend the interval
+ * to include the last {credIntervalCount} weeks.
  */
 export function immediateReceipts(
   budget: G.Grain,
-  identities: ProcessedIdentities
+  identities: ProcessedIdentities,
+  credIntervalCount: number = 1
 ): $ReadOnlyArray<GrainReceipt> {
-  const amounts = G.splitBudget(
-    budget,
-    identities.map((i) => i.mostRecentCred)
+  const totalIntervals = identities[0].cred.length;
+  if (credIntervalCount < 1 || credIntervalCount > totalIntervals) {
+    throw new Error(
+      `credIntervalCount must be in range [1, totalIntervals], got ${credIntervalCount}`
```

Comment:
I think failing when `intervalCount > totalIntervals` is annoying, b.c. it means a config that is valid for one project might be invalid for another. E.g. I decide to fork off of MetaGame's config, but they have a policy that distributes for 12 weeks lookback, but my project is only 4 weeks old. I recommend soft capping the interval count instead.

Reply:
Good point.